src/RemoteRequest/CachedRemoteGetRequest.php

```diff
@@ -111,7 +111,7 @@ public function get( $url ) {
 		$cached_response = get_transient( $cache_key );
 		$headers         = [];
 
-		if ( false !== $cached_response ) {
+		if ( is_string( $cached_response ) ) {
```

Comment:
> Seems the issue lies within `get_option()` unserializing the cached response before returning it. I think checking if the `$cached_response` is a string before attempting to unserialize it would be sufficient then.

I'm trying to understand how this was originally written. It was previously trying to check if `$cached_response` was set, as `get_transient()` returns `false` if the transient doesn't exist. 

However, if the transient _was_ set then it was expected that the return value would be a string and not the unserialized object. 

What's strange to me is that this method below actually sets it to a serialized string:

```php
set_transient( $cache_key, serialize( $cached_response ), $transient_expiry->getTimestamp() )
```

Nevertheless, `set_transient()` automatically will do such serialization via `maybe_serialize()`, at least when I see that options are being used as the storage backend. It seems persistent object cache implementations would also serialize to a string for storage. 

In that case, the `serialize()` in this method is redundant, no? I guess it's in order to have the improved security of the second argument to `unserialize()`, and that's why manual serialization/unserialization is being done.

In any case, `get_transient()` should be already also doing a `maybe_unserialize()`. Maybe the issue is with a persistent object cache that is doing the unserialization to `CachedResponse` eagerly? Or maybe a persistent object cache isn't doing the re-serialization of pre-serialized data? Notice how `maybe_serialize()` includes this logic:

```php
    /*
     * Double serialization is required for backward compatibility.
     * See https://core.trac.wordpress.org/ticket/12930
     * Also the world will end. See WP 3.6.1.
     */
    if ( is_serialized( $data, false ) ) {
        return serialize( $data );
    }
```

On my test environment, `$cached_response` here is a serialized string.

Reply:
In any case, checking for a string here looks good. Ref: https://3v4l.org/63IvG

![image](https://user-images.githubusercontent.com/134745/127791872-08f7fe98-f306-4913-9f0c-770207acb871.png)

Reply:
> Maybe the issue is with a persistent object cache that is doing the unserialization to `CachedResponse` eagerly?

I'm wondering if that's the case as well. Or perhaps the object cache was previously configured to store transients in the DB, thus skipping the re-serializaion, and later the object cache was  disabled/removed. That would mean when it fetches the transient and unserializes it the `CachedResponse` object would be returned.